omegaconf/dictconfig.py

```diff
@@ -47,13 +48,13 @@
 from .nodes import EnumNode, ValueNode
 
 
-class DictConfig(BaseContainer, MutableMapping[str, Any]):
+class DictConfig(BaseContainer, MutableMapping[DictKeyType, Any]):
```

Comment:
The following is giving me issues with mypy:
```
from omegaconf import OmegaConf
cfg = OmegaConf.create()
upd = dict(a=1)
cfg.update(upd)
```
The mypy error is:
```
tmp.py:4: error: Argument 1 to "update" of "MutableMapping" has incompatible type "Dict[str, int]"; expected "Mapping[Union[str, int, Enum], Any]"
```
This makes me think that

    class DictConfig(BaseContainer, MutableMapping[DictKeyType, Any]): ...


should be changed to

    class DictConfig(BaseContainer, MutableMapping[Any, Any]): ...

Reply:
Made the change in commit https://github.com/omry/omegaconf/pull/454/commits/3b193b601a43302a2c5d6107ef2df0afa744992f

Reply:
Unfortunately I don't really see a better option. I wonder if this is actually a bug in mypy?
(another "fix" would be to explicitly declare `upd: Dict[DictKeyType, Any] = dict(a=1)` but I think your solution is better)
@omry what do you think?

Reply:
I am okay with making DictConfig extend MutableMapping[Any, Any].

Reply:
> I am okay with making DictConfig extend MutableMapping[Any, Any].

Perfect, resolving then